src/condor_shadow.V6.1/shadow_v61_main.cpp

```diff
@@ -196,6 +196,15 @@ readJobAd( void )
 			EXCEPT( "Failed to insert \"%s\" into ClassAd!", line.c_str() );
         }
     }
+
+	// If we are a shadow running under a schedd, expect only one
+	// classad on stdin, and close it, to free the pipe and
+	// the resources associated with it.
+
+	if (sendUpdatesToSchedd) {
+		fclose(fp);
```

Comment:
```suggestion
		fclose(fp);
		fp = nullptr;
```
This make it clearer that another call to readJobAd() after the fclose() is invalid.

Reply:
If you want to get fancy, you can also have the function early-out if fp is nullptr.